cmd/ztest/ztest.c

```diff
@@ -6839,6 +6839,23 @@ ztest_run(ztest_shared_t *zs)
 	}
 	zs->zs_enospc_count = 0;
 
+	/*
+	 * If we were in the middle of ztest_device_removal() and were killed
+	 * we need to ensure the removal and scrub complete before running
+	 * any tests that check ztest_device_removal_active. The removal will
+	 * be restarted automatically when the spa is opened, but we need to
+	 * initate the scrub manually if it not already in progress.
+	 */
+	if (spa->spa_removing_phys.sr_state == DSS_SCANNING ||
+	    spa->spa_removing_phys.sr_prev_indirect_vdev != UINT64_MAX) {
```

Comment:
nit: perhaps we should check against `-1` instead to be consistent with the rest of the code and comment.

Reply:
Yes, since it looks like everything else `-1` is used.